### DISCUSSION TITLE
Orientation of AbaqusUMATStress

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#21376] [Q&A Modules: Solid mechanics] [answered]
If my UMAT has an anisotropic material, is there a way to re-orient it and/or the inputs going into it? 

tagging @dschwen (sorry for another UMAT question)

**Comment by recuero** [comment, marked as answer]
What is it in particular that you want to rotate/reorient?

**Reply by jessecarterMOOSE** [reply]
For my application I'd like to orient the material model a certain way. An example would be two grains that are the same material but are oriented differently in the global coordinate system. Ideally I'd use the same UMAT for both materials but MOOSE would handle the rotations somehow.

**Reply by recuero** [reply]
Doing that will most likely require some development around UMAT. MOOSE does have grain tracking capabilities that could probably be leveraged for your use case.

**Reply by jessecarterMOOSE** [reply]
Are there any existing MOOSE capabilities that could be strung together currently to do what I need? Doesn't have to be grains (that was just an example).

**Reply by recuero** [reply]
As far as I can see, code development will be required to allow the UMAT interface to correctly interpret local element/grain orientations. I assume you could potentially do what you want to do from the UMAT routines if you want to give it a try. But I understand that having it in the interface would be optimal.

**Reply by jessecarterMOOSE** [reply]
Does it make sense to apply a rotation to the stress/strain tensors before/after the call to the UMAT to represent some arbitrary orientation?

Unfortunately I can't touch the UMAT source code. Abaqus allows the user to specify an orientation to material models in the input file, which like you say is pretty convenient.
